thunder/executors/nvfuserex_impl.py

```diff
@@ -413,6 +415,31 @@ def __call__(self, *args):
     def __repr__(self):
         return f"FusionDefinitionWrapper({self.name})"
 
+    def last_inputs(self) -> str:
```

Comment:
+ @kevinstephano to consider if nvfuser itself would like to expose something like this. I think we discussed this briefly before, but maybe we could query nvFuser's Python API for something like "last reproduction" and that would include this information? Maybe the reproduction could be structured so it could be a Python string + contain structured information about the inputs?

For `last_inputs` here, I think we should consider just saving non-tensor inputs and only saving the metadata of tensor inputs. Non-tensor inputs can only be numbers or sequences of numbers, right? It'd be nice if last_inputs would return a valid set of inputs to the fusion, even creating the requested tensors based on the metadata. We could also rename this function to something like `print_last_inputs`.

I'm OK with this logic being in Thunder for now if it's helpful, even if we decide to expose it through nvfuser. When the nvfuser exposure is available we can remove this. Let's just add a warning when it's used that it's experimental and may change in the future.

How does that sound, @riccardofelluga? @IvanYashchuk? @kevinstephano?

Reply:
Yes, I agree that it should be possible to query from nvFuser's Python API the metadata of the last executed inputs.

I like the `print_last_inputs` name more, the current name suggests like real last inputs are captured, which is not the case. Currently, only tensor metadata is printed here, but nvFuser fusions also accept numbers.